pyppeteer/jshandle.py

```diff
@@ -0,0 +1,613 @@
+import asyncio
+import copy
+import math
+import os
+from typing import Dict, Optional, List, Any
+
+from pyppeteer import helper
+from pyppeteer.errors import BrowserError, ElementHandleError
+from pyppeteer.helper import debugError
+from pyppeteer.util import merge_dict
+
+
+def createJSHandle(context, remoteObject):
+    frame = context.frame()
+    if remoteObject.get('subtype') == 'node' and frame:
+        frameManager = frame._frameManager
+        return ElementHandle(
+            context, context._client, remoteObject, frameManager.page(), frameManager
+        )
+    return JSHandle(
+        context,
+        context._client,
+        remoteObject,
+    )
+
+
+class JSHandle(object):
+    """JSHandle class.
+
+    JSHandle represents an in-page JavaScript object. JSHandle can be created
+    with the :meth:`~pyppeteer.page.Page.evaluateHandle` method.
+    """
+
+    def __init__(
+            self,
+            context: 'ExecutionContext',
+            client: 'CDPSession',
+            remoteObject: 'RemoteObject'
+    ):
+        self._context = context
+        self._client = client
+        self._remoteObject = remoteObject
+        self._disposed = False
+
+    @property
+    def executionContext(self):
+        """Get execution context of this handle."""
+        return self._context
+
+    async def evaluate(self, pageFunction: str, *args):
+        return await self.executionContext.evaluate(pageFunction, *args)
+
+    async def evaluateHandle(self, pageFunction: str, *args):
+        return await self.executionContext.evaluateHandle(pageFunction, *args)
+
+    async def getProperty(self, propertyName: str) -> 'JSHandle':
+        """Get property value of ``propertyName``."""
+        objectHandle = await self._context.evaluateHandle(
+            '''(object, propertyName) => {
+                const result = {__proto__: null};
+                result[propertyName] = object[propertyName];
+                return result;
+            }''', self, propertyName)
+        properties = await objectHandle.getProperties()
+        result = properties[propertyName]
+        await objectHandle.dispose()
+        return result
+
+    async def getProperties(self) -> Dict[str, 'JSHandle']:
+        """Get all properties of this handle."""
+        response = await self._client.send('Runtime.getProperties', {
+            'objectId': self._remoteObject.get('objectId', ''),
+            'ownProperties': True,
+        })
+        result = dict()
+        for prop in response['result']:
+            if not prop.get('enumerable'):
+                continue
+            result[prop.get('name')] = createJSHandle(self._context, prop.get('value'))
```

Comment:
since we know the name attr is going to be specified, I think we can just directly access the name key in both cases
```suggestion
            result[prop['name']] = createJSHandle(self._context, prop['value'])
```